js/main.js

```diff
@@ -201,7 +209,162 @@ var renderCard = function (element) {
+};
+
+// module4-task2
+
+// Добавляем атрибут disabled для Fieldset-ов
+var formHeader = document.querySelector('.ad-form-header');
+formHeader.setAttribute('disabled', 'disabled');
+var formElement = document.querySelectorAll('.ad-form__element');
+for (var i = 0; i < formElement.length; i++) {
+  var formItem = formElement[i];
+  formItem.setAttribute('disabled', 'disabled');
+}
+
+// Функция активации сценария
+var activeForm = function () {
+  map.classList.remove('map--faded'); // Убираем класс-модификатор map--faded
+  document.querySelector('.ad-form').classList.remove('ad-form--disabled'); // Убираем класс-модификатор ad-form--disabled
+  renderPins(); // Функция создания 8 случайных пинов
+  cardCreate(); // Функция создания карточки дл первого пина
+  formHeader.removeAttribute('disabled', 'disabled');
+  formElement = document.querySelectorAll('.ad-form__element');
+  addressPin.value = coordinateMainPinActive();
+  for (var j = 0; j < formElement.length; j++) {
+    formItem = formElement[j];
+    formItem.removeAttribute('disabled', 'disabled');
+  }
+};
+
+// Выполнение сценария по нажатию на левую клавишу мыши
+var pinMain = document.querySelector('.map__pin--main');
+pinMain.addEventListener('mousedown', function (evt) {
+  if (evt.button === MOUSE_LEFT_BTN) {
+    activeForm();
+  }
+});
+
+// Выполнение сценария по нажатию на Enter
+pinMain.addEventListener('keydown', function (evt) {
+  if (evt.keyCode === ENTER_BUTTON) {
+    activeForm();
+  }
+});
+
+// Функция указания адреса главной круглой метки в неактивном состоянии (центр круглой метки)
+var coordinateMainPinInactive = function () {
+  var pinMainX = Number(parseInt(pinMain.style.left, RADIX)) + PIN_MAIN_SHIFT_X;
+  var pinMainY = Number(parseInt(pinMain.style.top, RADIX)) + PIN_MAIN_SHIFT_Y;
+  return pinMainX + ', ' + pinMainY;
+};
+
+// Функция указания адреса главной круглой метки в неактивном состоянии (центр круглой метки)
+var coordinateMainPinActive = function () {
+  var pinMainX = Number(parseInt(pinMain.style.left, RADIX)) + PIN_MAIN_SHIFT_X;
+  var pinMainY =
+    Number(parseInt(pinMain.style.top, RADIX)) +
+    PIN_MAIN_HEIGHT +
+    PIN_MAIN_POINT_SHIFT_Y;
+  return pinMainX + ', ' + pinMainY;
+};
+
+// Заполнение адрессной строки по умолчанию (в неактивном состоянии)
+var addressPin = document.querySelector('#address');
+addressPin.value = coordinateMainPinInactive();
 
-var mapFiltersContainer = map.querySelector('.map__filters-container');
-map.insertBefore(card, mapFiltersContainer);
+// Валидация формы ввода
+
+// // Валидация заголовка жилья
+var addForm = document.querySelector('.ad-form');
+var formTitleInput = addForm.querySelector('#title');
+
+formTitleInput.addEventListener('invalid', function () {
+  if (formTitleInput.validity.tooShort) {
+    formTitleInput.setCustomValidity(
+        'Заголовок не должен быть меньше ' + MIN_TITLE_LENGTH + '-ти символов'
```

Comment:
```suggestion
        'Заголовок не должен быть меньше ' + MIN_TITLE_LENGTH + '-ти символов'
```
А если заголовок будет из 8 символов, то будет "меньше 7-ти символов"?